var/spack/repos/builtin/packages/molcas/package.py

```diff
@@ -0,0 +1,57 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+
+
+class Molcas(CMakePackage):
+    """Molcas is an ab initio quantum chemistry software package
+       developed by scientists to be used by scientists.
+       Please set the path to licence file with the following command
+       export MOLCAS_LICENSE=/path/to/molcas/license/"""
+
+    homepage = "http://www.molcas.org/"
+
+    version('8.2', '25b5fb8e1338b458a3eaea0b3d3b5e58')
+
+    # Licensing
+    license_required = True
+    license_comment = ''
+    licnese_file = ['license.dat']
+    license_vars = ['MOLCAS_LICENSE']
+
+    depends_on('openmpi')
+    depends_on('openblas')
+    depends_on('hdf5')
+
+    patch('install_driver.patch')
+
+    def url_for_version(self, version):
+        return "file://{0}/molcas{1}.tar.gz".format(
+            os.getcwd(), version)
```

Comment:
Even if you add new versions, Spack will take this URL, detect the version (8.2), and replace it with the version you request. This is how all packages work. `url_for_version` is only needed in extreme circumstances.